hollow/src/main/java/com/netflix/hollow/api/producer/AbstractHollowProducer.java

```diff
@@ -388,6 +388,7 @@ long runCycle(
                 } else {
                     writeEngine.getHeaderTags().remove(HollowStateEngine.HEADER_TAG_SCHEMA_CHANGE);
                 }
+                writeEngine.addHeaderTag(HollowStateEngine.HEADER_TAG_PRODUCER_TO_VERSION, String.valueOf(toVersion));
```

Comment:
The reverse delta header is a special case. When running cycle v2 we want the reverse delta header to have
```hollow.producer.to.version=v1```
and I think that should be getting handled already [here](https://github.com/Netflix/hollow/blob/ec5dcb4fe45385ddac8c906a4ee0a2debe9ed6f9/hollow/src/main/java/com/netflix/hollow/core/write/HollowBlobWriter.java#L296-L297) but its worth testing this out and if possible write a unit test, maybe [here](hollow/src/test/java/com/netflix/hollow/core/write/HollowWriteStateEngineTest.java)